hearthstone/simulator/core/discover_object.py

```diff
@@ -0,0 +1,31 @@
+import itertools
+import typing
+from typing import Union, List, Callable
+
+from hearthstone.simulator.core.cards import MonsterCard
+from hearthstone.simulator.core.hero import Hero
+from hearthstone.simulator.core.secrets import Secret
+from hearthstone.simulator.core.spell import Spell
+
+if typing.TYPE_CHECKING:
+    from hearthstone.simulator.core.player import DiscoverIndex, Player
+
+Discoverable = Union[MonsterCard, Hero, Spell, Secret]
+
+
+class DiscoverObject:
+    def __init__(self, items: List[Discoverable], discover_function: Callable[[Discoverable], None], dissolve_leftovers: bool):
+        self.items = items
+        self.discover_function = discover_function
+        self.dissolve_leftovers = dissolve_leftovers
+
+    def select_item(self, index: 'DiscoverIndex', player: 'Player'):
+        selected = self.items.pop(index)
+        if isinstance(selected, MonsterCard):
+            selected.token = False  # for Bigglesworth (there is no other scenario where a token will be a discover option)
```

Comment:
What does setting `token=False` do here?

Reply:
If I'm remembering correctly, if Bigglesworth discovers a token, once sold that token is actually added to the pool, which does not happen normally when a token is sold.

Reply:
I see.